### PR TITLE
Fix migrate use after free

### PR DESCRIPTION
**Pull request by shinrich**
Another ASAN issue found by @bneradt in our prod sim environment.

In this case the original thread handles the delete of the netvc object while the new thread is still using it.  By moving the do_io_close to the end, the original thread will not have the chance to delete until after the new thread is done with the object.

```
==118316==ERROR: AddressSanitizer: heap-use-after-free on address 0x61a0000ad01c at pc 0x000000ee0719 bp 0x2b52bfe67ed0 sp 0x2b52bfe67ec0
READ of size 4 at 0x61a0000ad01c thread T16 ([ET_NET 14])
    #0 0xee0718 in UnixNetVConnection::migrateToCurrentThread(Continuation*, EThread*) ../../../../../../_vcs/trafficserver9/iocore/net/UnixNetVConnection.cc:1400
    #1 0x852720 in HttpSessionManager::acquire_session(Continuation*, sockaddr const*, char const*, ProxyTransaction*, HttpSM*) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSessionManager.cc:346
    #2 0x82864a in HttpSM::do_http_server_open(bool) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:4902
    #3 0x833c1d in HttpSM::set_next_state() ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:7428
    #4 0x7da73c in HttpSM::call_transact_and_set_next_state(void (*)(HttpTransact::State*)) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:7231
    #5 0x81e559 in HttpSM::handle_api_return() ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:1603
    #6 0x80e595 in HttpSM::state_api_callout(int, void*) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:1535
    #7 0x819f72 in HttpSM::state_api_callback(int, void*) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:1339
    #8 0x6dd5c4 in TSHttpTxnReenable ../../../../../_vcs/trafficserver9/src/traffic_server/InkAPI.cc:6097
    #9 0x2b52c291ba5c in cb _vcs/quick_filter-9/quick_filter/quick_filter.cc:627


...


0x61a0000ad01c is located 412 bytes inside of 1152-byte region [0x61a0000ace80,0x61a0000ad300)
freed by thread T19 ([ET_NET 17]) here:
    #0 0x61bb10 in free (/opt/oath/trafficserver/9.0/bin/traffic_server+0x61bb10)
    #1 0x2b52b486789c in ink_freelist_free ../../../../../../_vcs/trafficserver9/src/tscore/ink_queue.cc:277
    #2 0xe98a12 in NetHandler::free_netvc(UnixNetVConnection*) ../../../../../../_vcs/trafficserver9/iocore/net/UnixNet.cc:353
    #3 0xe99402 in NetHandler::process_ready_list() ../../../../../../_vcs/trafficserver9/iocore/net/UnixNet.cc:399
    #4 0xe9a296 in NetHandler::waitForActivity(long) ../../../../../../_vcs/trafficserver9/iocore/net/UnixNet.cc:536
    #5 0xfc44ae in EThread::execute_regular() ../../../../../../_vcs/trafficserver9/iocore/eventsystem/UnixEThread.cc:271
    #6 0xfc4d05 in EThread::execute() ../../../../../../_vcs/trafficserver9/iocore/eventsystem/UnixEThread.cc:332
    #7 0xfc4d05 in EThread::execute() ../../../../../../_vcs/trafficserver9/iocore/eventsystem/UnixEThread.cc:310
    #8 0xfbe7ea in spawn_thread_internal ../../../../../../_vcs/trafficserver9/iocore/eventsystem/Thread.cc:92
    #9 0x2b52b6230dd4 in start_thread (/lib64/libpthread.so.0+0x7dd4)

previously allocated by thread T19 ([ET_NET 17]) here:
    #0 0x61ccf0 in posix_memalign (/opt/oath/trafficserver/9.0/bin/traffic_server+0x61ccf0)
    #1 0x2b52b486578f in ats_memalign ../../../../../../_vcs/trafficserver9/src/tscore/ink_memory.cc:102
    #2 0x2b52b48677fe in ink_freelist_new ../../../../../../_vcs/trafficserver9/src/tscore/ink_queue.cc:187
    #3 0xe26340 in ClassAllocator<SSLNetVConnection>::alloc() /home/bneradt/repos/build_ts_9/_build/build_release_posix-x86_64/trafficserver9/build/../../../../_vcs/trafficserver9/include/tscore/Allocator.h:131
    #4 0xe26340 in SSLNetVConnection* thread_alloc_init<SSLNetVConnection>(ClassAllocator<SSLNetVConnection>&, ProxyAllocator&) /home/bneradt/repos/build_ts_9/_build/build_release_posix-x86_64/trafficserver9/build/../../../../_vcs/trafficserver9/iocore/eventsystem/I_ProxyAllocator.h:73
    #5 0xe26340 in SSLNetProcessor::allocate_vc(EThread*) ../../../../../../_vcs/trafficserver9/iocore/net/SSLNetProcessor.cc:104
    #6 0xee0384 in UnixNetVConnection::migrateToCurrentThread(Continuation*, EThread*) ../../../../../../_vcs/trafficserver9/iocore/net/UnixNetVConnection.cc:1395
    #7 0x852720 in HttpSessionManager::acquire_session(Continuation*, sockaddr const*, char const*, ProxyTransaction*, HttpSM*) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSessionManager.cc:346
    #8 0x82864a in HttpSM::do_http_server_open(bool) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:4902
    #9 0x833c1d in HttpSM::set_next_state() ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:7428
    #10 0x7da73c in HttpSM::call_transact_and_set_next_state(void (*)(HttpTransact::State*)) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:7231
    #11 0x81e559 in HttpSM::handle_api_return() ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:1603
    #12 0x80e595 in HttpSM::state_api_callout(int, void*) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:1535
    #13 0x819f72 in HttpSM::state_api_callback(int, void*) ../../../../../../_vcs/trafficserver9/proxy/http/HttpSM.cc:1339
    #14 0x6dd5c4 in TSHttpTxnReenable ../../../../../_vcs/trafficserver9/src/traffic_server/InkAPI.cc:6097
    #15 0x2b52c291ba5c in cb _vcs/quick_filter-9/quick_filter/quick_filter.cc:627
```

### CODE REVIEW

**Comment by shinrich** [comment]
[approve ci clang-analyzer]

**Comment by shinrich** [comment]
[approve ci autest]

File: iocore/net/UnixNetVConnection.cc [hunk 1]
```diff
@@ -1389,19 +1389,19 @@ UnixNetVConnection::migrateToCurrentThread(Continuation *cont, EThread *t)
...
       sslvc = nullptr;
     } else {
+      // Update the SSL fields
       sslvc->set_context(get_context());
     }
-    return sslvc;
-    // Update the SSL fields
+    netvc = sslvc;
   } else {
     UnixNetVConnection *netvc = static_cast<UnixNetVConnection *>(netProcessor.allocate_vc(t));
```
Comment (masaori335) [inline iocore/net/UnixNetVConnection.cc]: Declaring `UnixNetVConnection *netvc` again.

Comment (shinrich) [reply iocore/net/UnixNetVConnection.cc]: Ah.  I'm blind.  Thanks for pointing out.

**Comment by randall** [comment]
[approve ci autest]

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**